docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_use_cdn.mdx

```diff
@@ -23,6 +23,281 @@ CBP
+
+   ```sh
+   aws cloudfront create-distribution --origin-domain-name your-origin-domain-name
+   ```
+
+2. **Specify the Origin:**
+   When creating the distribution, specify the origin (e.g., your S3 bucket or your web server) that CloudFront will use to fetch the content.
+
+   ```sh
+   aws cloudfront create-distribution --origin-domain-name your-origin-domain-name --default-root-object index.html
+   ```
+
+3. **Configure Cache Behaviors:**
+   Define cache behaviors to specify how CloudFront will cache your content. This can be done during the creation of the distribution or by updating an existing distribution.
+
+   ```sh
+   aws cloudfront create-distribution --origin-domain-name your-origin-domain-name --default-root-object index.html --cache-behaviors Quantity=1,Items=[{PathPattern='*',TargetOriginId='your-origin-id',ViewerProtocolPolicy='redirect-to-https'}]
+   ```
+
+4. **Enable Distribution:**
+   Ensure that the distribution is enabled so that it starts serving your content via the CDN.
+
+   ```sh
+   aws cloudfront create-distribution --origin-domain-name your-origin-domain-name --default-root-object index.html --enabled
+   ```
+
+By following these steps, you can set up a CloudFront distribution using the AWS CLI, ensuring that your web applications use CDNs to improve performance and reliability.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of web applications not using CDNs in CloudFront using Python scripts, you can follow these steps:
+
+1. **Install Required Libraries**:
+   Ensure you have the `boto3` library installed, which is the AWS SDK for Python. This library allows you to interact with AWS services.
+
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+   ```bash
+   aws configure
+   ```
+
+   Alternatively, you can set environment variables:
+
+   ```bash
+   export AWS_ACCESS_KEY_ID='your_access_key_id'
+   export AWS_SECRET_ACCESS_KEY='your_secret_access_key'
+   export AWS_DEFAULT_REGION='your_default_region'
+   ```
+
+3. **Create a Python Script to Check and Create CloudFront Distribution**:
+   Write a Python script that checks if a CloudFront distribution exists for your web application. If it doesn't, the script will create one.
+
+   ```python
+   import boto3
+
+   def create_cloudfront_distribution(origin_domain_name):
+       client = boto3.client('cloudfront')
+
+       response = client.create_distribution(
+           DistributionConfig={
+               'CallerReference': 'unique-string',
+               'Origins': {
+                   'Quantity': 1,
+                   'Items': [
+                       {
+                           'Id': '1',
+                           'DomainName': origin_domain_name,
+                           'OriginPath': '',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>